services/graphql/src/resolvers/customer.js

```diff
@@ -336,29 +337,27 @@ module.exports = {
         return map;
       }, deploymentTypeIdMap);
 
+      // Append newsletter product subscriptions for each opted-in email deployment
       const deploymentTypeIds = [...deploymentTypeOptInMap.keys()];
-
-      const productDeploymentTypeMap = await (async () => {
-        if (!deploymentTypeIds.length) return [];
+      if (deploymentTypeIds.length) {
         const cursor = await repos.brandProduct.find({
           query: {
             'data.DeploymentTypeId': { $in: deploymentTypeIds },
             'data.ProductType': 2, // only newsletters can be subscribed to in this manner
           },
           options: { projection: { 'data.Id': 1, 'data.DeploymentTypeId': 1 } },
         });
-        const docs = await cursor.toArray();
-        return docs.reduce((map, doc) => {
+        await cursor.forEach((doc) => {
           const { Id, DeploymentTypeId } = doc.data;
-          map.set(Id, DeploymentTypeId);
-          return map;
-        }, new Map());
-      })();
+          const optedIn = deploymentTypeOptInMap.get(DeploymentTypeId);
+          if (optedIn == null) return;
+          Products.push({ OmedaProductId: Id, Receive: Number(optedIn) });
+        });
+      }
 
-      productDeploymentTypeMap.forEach((deploymentTypeId, productId) => {
-        const optedIn = deploymentTypeOptInMap.get(deploymentTypeId);
-        if (optedIn == null) return;
-        Products.push({ OmedaProductId: productId, Receive: Number(optedIn) });
+      // Append explicitly provided product subscriptions
+      subscriptions.forEach(({ id, receive }) => {
+        Products.push({ OmedaProductId: id, Receive: Number(receive) });
```

Comment:
Account for newsletter product types of `2` and add to the deployment type optin/optout map (if not already present in the map)